pds_doi_service/core/actions/test/check_test.py

```diff
@@ -66,26 +76,166 @@ def webclient_query_patch(self, i_url, query_dict=None, i_username=None,
+
+        self.assertEqual(pending_record['previous_status'], DoiStatus.Pending)
+        self.assertEqual(pending_record['status'], DoiStatus.Error)
+        self.assertEqual(pending_record['submitter'], 'img-submitter@jpl.nasa.gov')
+        self.assertEqual(pending_record['doi'], '10.17189/29348')
+        self.assertEqual(pending_record['lidvid'], 'urn:nasa:pds:lab_shocked_feldspars::1.0')
+
+        # There should be a message to go along with the error
+        self.assertIsNotNone(pending_record['message'])
+
+    @patch.object(
+        pds_doi_service.core.outputs.osti_web_client.DOIOstiWebClient,
+        'webclient_query_doi', webclient_query_patch_no_change)
+    def test_check_for_pending_entries_w_no_change(self):
+        """Test check action when no pending entries have been updated"""
+        pending_records = self._action.run(email=False)
+
+        self.assertEqual(len(pending_records), 1)
+
+        pending_record = pending_records[0]
+
+        self.assertEqual(pending_record['previous_status'], DoiStatus.Pending)
+        self.assertEqual(pending_record['status'], DoiStatus.Pending)
+        self.assertEqual(pending_record['submitter'], 'img-submitter@jpl.nasa.gov')
+        self.assertEqual(pending_record['doi'], '10.17189/29348')
+        self.assertEqual(pending_record['lidvid'], 'urn:nasa:pds:lab_shocked_feldspars::1.0')
+
+    def get_config_patch(self):
+        """
+        Return a modified default config that points to a local test smtp
+        server for use with the email test
+        """
+        parser = configparser.ConfigParser()
+
+        # default configuration
+        conf_default = 'conf.ini.default'
+        conf_default_path = abspath(
+            join(dirname(__file__), os.pardir, os.pardir, 'util', conf_default)
+        )
+
+        parser.read(conf_default_path)
+        parser['OTHER']['emailer_local_host'] = 'localhost'
+        parser['OTHER']['emailer_port'] = '1025'
+
+        parser = DOIConfigUtil._resolve_relative_path(parser)
+
+        return parser
+
+    @patch.object(
+        pds_doi_service.core.util.config_parser.DOIConfigUtil,
+        'get_config', get_config_patch)
+    @patch.object(
+        pds_doi_service.core.outputs.osti_web_client.DOIOstiWebClient,
+        'webclient_query_doi', webclient_query_patch_nominal)
+    def test_email_receipt(self):
+        """Test sending of the check action status via email"""
+        # Create a new check action so our patched config is pulled in
+        action = DOICoreActionCheck(self.db_name)
+
+        with tempfile.TemporaryFile() as temp_file:
+            # Stand up a subprocess running a debug smtpd server
+            # By default, all this server is does is echo email payloads to
+            # standard out, so provide a temp file to capture it
+            debug_email_proc = subprocess.Popen(
+                ['python', '-u', '-m', 'smtpd', '-n', '-c', 'DebuggingServer', 'localhost:1025'],
```

Comment:
`smtpd` module?! Oh wow, I had no idea this was even part of the Python Standard Library! ✉️

Cool! 😎